colossus/src/main/scala/colossus/core/WorkerItem.scala

```diff
@@ -105,3 +113,47 @@ abstract class WorkerItem(val context: Context) {
 
 
 }
+
+trait ProxyActor { self: WorkerItem => 
+  
+  import ProxyActor._
+  
+  implicit lazy val self : ActorRef = context.proxy
+
+  private var currentReceiver : Receive = receive
+    
+  def becomeReceive(receive: Receive) {
+    currentReceiver = receive
+  }
+  
+  private var lastSender = ActorRef.noSender
+  def sender() : ActorRef = lastSender
+
+  def receivedMessage(message: Any, sender: ActorRef) {
+    lastSender = sender
+    currentReceiver.applyOrElse(message, (_: Any) => ())
```

Comment:
Yeah I was thinking that as well, I think we can actually just use Akka's deadletters.